### PR TITLE
Add a lua function to query the terrain type of a cell

### PR DESCRIPTION
**Pull request by abcdefg30**

### CODE REVIEW

File: OpenRA.Mods.Common/Scripting/Global/MapGlobal.cs [hunk 1]
```diff
@@ -101,6 +101,12 @@ public WPos CenterOfCell(CPos cell)
 			return Context.World.Map.CenterOfCell(cell);
 		}
 
+		[Desc("Returns the type of the tile at the target cell.")]
+		public string TileType(CPos cell)
```
Comment (pchote) [inline OpenRA.Mods.Common/Scripting/Global/MapGlobal.cs]: This is actually returning the Terrain Type, not the Tile Type.  Can you please rename it?

Comment (abcdefg30) [inline OpenRA.Mods.Common/Scripting/Global/MapGlobal.cs]: Updated.

**Comment by Phrohdoh** [comment]
Why is this desired?
Is there some other data that someone _actually_ wants?

**Comment by ltem** [comment]
IRC_LOG_2017-02-12
>[16:46:38]	<SirCake> abcdefg30: do you know a way to check which terrain tile is at a given position? I am trying to make a lua check if my unit is on rock terrain or not.
>[16:46:54]	<abcdefg30> sorry, I don't think that is possible
>[16:47:14]	<abcdefg30> (only by having a table { } in lua with each tile type per position)
>[16:47:17]	<SirCake> dang, but the info has to be in the map files ...
>[16:47:19]	<abcdefg30> (but that'd be a PITA to do)

@Phrohdoh and this resulted into this PR :)
A usecase could be that the AI prefers to position its infantry on rock terrain, if it is fighting against tanks

**Comment by GraionDilach** [comment]
It can also be useful for potential waypoint paths with mixed taskforces (where amphibious units are along with usual ground ones etc.), or checking if the ice has regrown on a TS map, or if a route has been concretized in Dune2000 or anything involving custom terrains.

**Review by pchote — APPROVED** [review]
Works as advertised

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/70ce49e3211f216e050d9f3c1689d212db94cb4b

**Pull request merged**